src/buildercore/s3.py

```diff
@@ -1,46 +1,59 @@
 import os
-import boto
-from boto import s3
-from boto.s3.key import Key
+import boto3
+from botocore.exceptions import ClientError
 from . import config
-from .utils import isstr
+from .utils import isstr, ensure
 from kids.cache import cache as cached
 from io import IOBase
 import logging
 
 LOG = logging.getLogger(__name__)
 
-def connect_s3():
-    # we'll need to deal with this assumption
-    return s3.connect_to_region('us-east-1')  # Location.USWest2)
+def connect_s3(region='us-east-1'):
+    return boto3.resource('s3', region_name=region)
 
 @cached
 def builder_bucket():
+    "returns connection to the bucket where builder stores templates and credentials."
     try:
-        nom = config.BUILDER_BUCKET
-        return connect_s3().get_bucket(nom)
-    except boto.exception.S3ResponseError as err:
-        LOG.error("got an S3 error attempting to get the builder bucket. have you created it yet?",
-                  extra={'bucket': nom, 'error': err.message})
+        nom, region = config.BUILDER_BUCKET, config.BUILDER_REGION
+        resource = connect_s3(region)
+        bucket = resource.Bucket(nom)
+        ensure(bucket in resource.buckets.all(), "bucket %r in region %r does not exist" % (nom, region))
+        return bucket
+    except ClientError as err:
+        LOG.error("unhandled error attempting to find S3 bucket %r in region %r", nom, region,
+                  extra={'bucket': nom, 'region': region, 'error': err.message})
         raise
 
 def exists(key):
-    return builder_bucket().get_key(key) is not None
+    "predicate, returns True if given key in configured bucket+region exists"
+    try:
+        builder_bucket().Object(key).load()
```

Comment:
I thought it could be checked without actually loading the value of the key (like an underlying `HEAD` request) but for this use case it shouldn't be a problem

Reply:
consider also https://stackoverflow.com/a/38376288/91590 if you haven't already excluded it

Reply:
already considered it ;) see docs: http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Object.load